addons/mail/controllers/discuss/channel.py

```diff
@@ -141,3 +141,27 @@ def discuss_channel_fold(self, channel_id, state, state_count):
         if not member:
             raise NotFound()
         return member._channel_fold(state, state_count)
+
+    @http.route("/discuss/channel/sub_channel/fetch", methods=["POST"], type="json", auth="public")
+    @add_guest_to_context
+    def discuss_channel_sub_channel_fetch(self, parent_channel_id, search_term=None, before=None, limit=30):
+        channel = request.env["discuss.channel"].search([("id", "=", parent_channel_id)])
+        if not channel._find_or_create_member_for_self():
+            raise NotFound()
+        domain = [("parent_channel_id", "=", channel.id)]
+        if before:
+            domain.append(("id", "<", before))
+        if search_term:
+            domain.append(("name", "ilike", search_term))
+        sub_channels = request.env["discuss.channel"].search(domain, order="id desc", limit=limit)
+        store = Store(sub_channels)
+        store.add(sub_channels._get_last_messages())
+        return store.get_result()
+
+    @http.route("/discuss/channel/sub_channel/join", methods=["POST"], type="json", auth="public")
+    @add_guest_to_context
+    def discuss_channel_sub_channel_join(self, sub_channel_id):
+        if sub_channel := request.env["discuss.channel"].browse(sub_channel_id).exists():
+            sub_channel._find_or_create_member_for_self()
+            return Store(sub_channel).get_result()
+        raise NotFound()
```

Comment:
How it is method related to a sub_channel in any way?

It looks like just a standard channel join, as it doesn't ever check parent/child fields.

Don't we want to restrict this method to a channel with the parent field set?
Or just a more standard route?

Reply:
Indeed, it could have been a route for the `add_members` method. Will do it.